services/products-service/test/catalogBatchProcess.test.ts

```diff
@@ -0,0 +1,80 @@
+import { handler as catalogBatchProcess } from '../lambdas/catalogBatchProcess';
+import { SQSEvent } from 'aws-lambda';
+
+jest.mock('@aws-sdk/client-sns', () => ({
+  SNSClient: jest.fn(() => ({
+    send: jest.fn().mockResolvedValue({}),
+  })),
+  PublishCommand: jest.fn(),
+}));
+jest.mock('@aws-sdk/client-dynamodb', () => ({
+  DynamoDB: jest.fn(),
+}));
+jest.mock('@aws-sdk/lib-dynamodb', () => ({
+  DynamoDBDocumentClient: {
+    from: jest.fn(() => ({
+      send: jest.fn().mockResolvedValue({}),
+    })),
+  },
+  TransactWriteCommand: jest.fn(),
+}));
+
+const sqsEvent = {
+  Records: [
+    {
+      body: JSON.stringify({
+        product_id: '1',
+        title: 'Product 1',
+        description: 'Description 1',
+        count: 10,
+        price: 100,
+      }),
+    },
+  ],
+};
+
+describe('catalogBatchProcess', () => {
+  it('should handle SQS event and execute transactions and SNS publish', async () => {
+    process.env.productsTableName = 'mockProductsTable';
+    process.env.stocksTableName = 'mockStocksTable';
+    process.env.createProductTopic = 'mockCreateProductTopic';
+
+    await catalogBatchProcess(sqsEvent as SQSEvent);
+
+    expect(
+      require('@aws-sdk/lib-dynamodb').DynamoDBDocumentClient.from
```

Comment:
As you mocked the module, it would be better to store link to used mock function (e.g. DynamoDBDocumentClient.from) and check it here and not require module.